Demo/Demo/ViewModels/CardVaultViewModel.swift

```diff
@@ -31,10 +31,14 @@ class CardVaultViewModel: VaultViewModel, CardVaultDelegate {
         return enabled
     }
 
-    func setUpTokenSuccessResult(vaultResult: CardPayments.CardVaultResult) {
+    func setUpdateTokenSuccessResult(vaultResult: CardPayments.CardVaultResult) {
```

Comment:
The function is just setting the state to update the view, so imo it's confusing to use "updateSetupToken" which is kind of name for our endpoint as well. And I like having separate functions, especially as we change the values we return to merchant.

Reply:
Gotcha, since this and the error case are doing something similar, we could consider doing something like this:
```swift
    func updateSetupTokenResponseState(vaultResult: CardVaultResult?, vaultError: CoreSDKError?) {
        DispatchQueue.main.async {
            if let vaultError {
                self.state.updateSetupTokenResponse = .error(message: vaultError.localizedDescription)
            } else if let vaultResult {
                self.state.updateSetupTokenResponse = .loaded(
                    UpdateSetupTokenResult(
                        id: vaultResult.setupTokenID,
                        status: vaultResult.status,
                        threeDSecureAttempted: vaultResult.threeDSecureAttempted
                    )
                )
            }
        }
    }
```

Reply:
The delegate functions either have result or error, and I think it's clearer the way it is. Not strongly tied to it though.

Reply:
Yes, but with both values optional in the helper method, we can pass down the appropriate data from the delegate. So the error delegate would send `updateSetupTokenResponseState(vaultResult: nil, vaultError: vaultError)` and the success delegate would send `updateSetupTokenResponseState(vaultResult: vaultResult, vaultError: nil)`.